src/Solr/Query/SearchQueryBuilder.php

```diff
@@ -236,4 +239,28 @@ protected function getEventDispatcher()
     {
         return $this->eventDispatcher;
     }
+
+    /**
+     * Don't use "E" notation for values lower than 0.00001 (i.e. use 0.0000001 instead of 1E-07)
+     * Taken from https://stackoverflow.com/a/10917464/3141504
+     *
+     * @param float $boost
+     * @return string
+     */
+    private function getNormalizedBoost($boost)
+    {
+        if ($boost == 0) {
+            return $boost;
+        }
+        if ($boost < 1) {
```

Comment:
is the differenciation necessary? Without trimming zeroes, the formatting below works for all numbers, especially big values >~1E14

Reply:
Differenciation removed. You are right.